### ISSUE TITLE
dnsdist 1.3.3: armhf test failures

### ISSUE THREAD

**Issue by tcely** [PowerDNS/pdns#7590]
- Program: dnsdist 
 - Issue type: Bug report 

### Short description

```
*** 3 failures are detected in the test module "unit"

test-dnsdistpacketcache_cc.cc(114): error: in "test_dnsdistpacketcache_cc/test_PacketCacheSimple": check matches == expected has failed [88902 != 99994]
test-dnsdistrules_cc.cc(90): error: in "dnsdistluarules_cc/test_MaxQPSIPRule": check removed == (total / scanFraction) + 1 has failed [4951 != 6554]
test-dnsdistrules_cc.cc(92): error: in "dnsdistluarules_cc/test_MaxQPSIPRule": check scanned == removed has failed [4952 != 4951]
```

### Environment

 - Operating system: Alpine Linux
 - Software version: edge
 - Software source: compiled from source

### Expected behaviour

Tests passed on other platforms

### Actual behaviour
3 tests failed

### Other information

https://cloud.drone.io/alpinelinux/aports/536/4/2

**Comment by Habbie** [comment]
Hi! Can I ask about your armhf builder? Is it possible it is very slow? We have a few tests that almost consistently show 'off' numbers in very slow environments. That's a bug, of course, but knowing that would be helpful.

**Comment by Habbie** [comment]
As a weak datapoint, i've let 'make check' in dnsdist master run overnight in a loop, on Debian 9 arm64, without any errors.

**Comment by Habbie** [comment]
The build log also has some interesting compiler warnings that we should look at besides the test failure.

**Comment by Habbie** [comment]
dnsdist master `make check` on the docker armhf/alpine:edge image passes for me (qemu-emulated in Docker for Mac), it does take several hours.

**Comment by rgacogne** [comment]
> Can I ask about your armhf builder? Is it possible it is very slow? We have a few tests that almost consistently show 'off' numbers in very slow environments. That's a bug, of course, but knowing that would be helpful.

In that case, though, none of the tests that are failing involve threads or possible race, so I'm worried we might have a real issue. I would be very interested in a way to reliably reproduce that issue.

**Comment by Habbie** [comment]
This might even be unrelated, but repeating `test_dnsdistpacketcache_cc/test_PacketCacheSimple` for hours and hours eventually gave me `test-dnsdistpacketcache_cc.cc(114): error: in "test_dnsdistpacketcache_cc/test_PacketCacheSimple": check matches == expected has failed [0 != 99934]`

**Comment by tcely** [comment]
> Hi! Can I ask about your armhf builder? Is it possible it is very slow? We have a few tests that almost consistently show 'off' numbers in very slow environments. That's a bug, of course, but knowing that would be helpful.

It's not terribly slow. For instance, that builder without unit tests enabled finished building both variants in under 5 minutes.

https://cloud.drone.io/alpinelinux/aports/716/4/2

**Comment by rgacogne** [comment]
Alright, thanks for the update! It seems that you have done some tests with #7592 applied, do you know if that helped?

**Comment by tcely** [comment]
> Alright, thanks for the update! It seems that you have done some tests with #7592 applied, do you know if that helped?

I did try it with a patched version based on #7592. It did not help enough. The lua flavor built and tested correctly, but the luajit flavor timed out.

https://cloud.drone.io/alpinelinux/aports/704/4/2

**Comment by tcely** [comment]
It might be helpful to know why the unit tests are so very slow on `armhf` but complete in a reasonable time on `armv7` and `aarch64` too.
